packages/presentation/lib/services/firebase_analytics.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:firebase_analytics/firebase_analytics.dart';
+
+class Analytics {
+  final FirebaseAnalytics analytics;
+
+  const Analytics(this.analytics);
+
+  analyticsFacebookClick() async {
+    await analytics.logEvent(name: 'on_facebook_click');
+  }
+
+  analyticsGoogleClick() async {
+    await analytics.logEvent(name: 'on_google_click');
+  }
+
+  analyticsLoginClick() async {
+    await analytics.logEvent(name: 'on_login_click');
+  }
```

Comment:
no need in creating separate method for each event. in this case our service will be too big. just create one method - logEvent and pass here different params